File: TareaHerencia/TareaGloria/Villano.cpp
#include <string>
#include "Persona.h"
#include "Villano.h"

Villano::Villano(string nickname, string superpoder,string nombre,int edad) : Persona(nombre,edad){
    this->nickname=nickname;
    this->superpoder=superpoder;
}

void Villano::presentacion(){
    cout<<"Soy "<<nickname<<" y mi poder es "<<superpoder<<endl;
}

string Villano::revelarNombre(){
    cout<<"Mi nombre es "<<darNombre()<<endl;
}

File: TareaHerencia/Solucion/Python/Villano.py
from Persona import *

class Villano(Persona):
    def __init__(self,name,age,nick,poder):
        super().__init__(name,age)
        self.nickname = nick
        self.power = poder

    def presentacion(self):
        print('Yo soy {} y mi poder es {}'.format(self.nickname,self.power))

    def revelarNombre(self):
        print("Mi verdadero nombre es {}".format(self.darNombre()))